SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -402,6 +403,31 @@ bool SaveSchemaDatabase(const FString& PackagePath)
 	SchemaDatabase->ComponentIdToClassPath = CreateComponentIdToClassPathMap();
 	SchemaDatabase->LevelComponentIds = LevelComponentIds;
 
+	// Generate hash
+	{
+		FString DescriptorPath = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("build/assembly/schema/schema.descriptor"));
+		IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+
+		TUniquePtr<IFileHandle> FileHandle(PlatformFile.OpenRead(DescriptorPath.GetCharArray().GetData()));
+		if (!FileHandle)
+		{
+			FMessageDialog::Debugf(FText::FromString(TEXT("Failed to open schema.descriptor generated by the schema compiler!")));
+			return false;
+		}
+
+		// Create our byte buffer
+		int64 FileSize = FileHandle->Size();
+		TUniquePtr<uint8[]> ByteArray(new uint8[FileSize]);
```

Comment:
Curious - do TUniquePtr work with array allocations? ie. will it call the delete[] operator?

Reply:
Yeah, they have a struct TDefaultDelete<T[]>specialization. Good catch though because I just assumed it was the same as std